frontend/reducers/now_playing_reducer.js

```diff
@@ -1,20 +1,46 @@
-import { RECEIVE_CURRENT_ARTIST } from "../actions/artist_actions";
+import { QUEUE_ARTIST, PLAY_ARTIST, PUSH_PLAY } from "../actions/now_playing_actions";
 import { TOGGLE_PLAY } from "../actions/now_playing_actions";
 
 const nowPlayingReducer = (
-	playState = { isPlaying: false, queue: [] },
+	playState = {
+		isPlaying: false,
+		queue: [],
+		queueSource: { sourceType: null, extractedUrlParams: null },
+	},
 	action
 ) => {
 	Object.freeze(playState);
 	let newPlayState = Object.assign({}, playState);
 	switch (action.type) {
-		case RECEIVE_CURRENT_ARTIST:
-			newPlayState.queue = [...action.allSongs];
-			return newPlayState;
 		case TOGGLE_PLAY:
 			if (newPlayState.queue?.length > 0)
 				newPlayState.isPlaying = !newPlayState.isPlaying;
 			return newPlayState;
+		case PUSH_PLAY:
+			if (newPlayState.queue?.length > 0) newPlayState.isPlaying = true;
+			return newPlayState;
+		case QUEUE_ARTIST:
+			newPlayState.queue.push(...action.allSongs);
```

Comment:
If you rename `allSongs` to something like `songsToQueue` you might be able to reuse the same action for anther view.

Reply:
This is a great idea, thank you!

Reply:
Hm, actually I will be using the `allSongs` property of any artist or album for "Add to playlist" as well, so I don't think I will specify it as `songsToQueue` at this time.